src/Android/Xamarin.Android/Samples/Data/ReadGeoPackage/ReadGeoPackage.cs

```diff
@@ -0,0 +1,338 @@
+
+            // Open the GeoPackage
+            GeoPackage myGeoPackage = await GeoPackage.OpenAsync(myGeoPackagePath);
+
+            // Get the read only list of GeoPackageRasters from the GeoPackage
+            IReadOnlyList<GeoPackageRaster> myReadOnlyListOfGeoPackageRasters = myGeoPackage.GeoPackageRasters;
+
+            // Loop through each GeoPackageRaster
+            foreach (GeoPackageRaster oneGeoPackageRaster in myReadOnlyListOfGeoPackageRasters)
+            {
+                // Create a RasterLayer from the GeoPackageRaster
+                RasterLayer myRasterLayer = new RasterLayer(oneGeoPackageRaster);
+
+                // Set the opacity on the RasterLayer to partially visible 
+                myRasterLayer.Opacity = 0.55;
+
+                // Load the RasterLayer - that way we can get to it's properties
+                await myRasterLayer.LoadAsync();
+
+                // Create a string variable to hold the human-readable name of the RasterLayer for display
+                // in the ListBox and the HybridDictonary - it will initially be an empty string
+                string myRasterLayerName = "";
+
+                if (myRasterLayer.Name != "")
+                {
+                    // We have a good human-readable name for the RasterLayer that came from
+                    // the RasterLayer.Name property
+                    myRasterLayerName = myRasterLayer.Name;
+                }
+                else if (oneGeoPackageRaster.Path.Split('/').Last() != "")
+                {
+                    // We did not get a good human-readable name from the RasterLayer from the .Name
+                    // property, get the good human-readable name from the GeoPackageRaster.Path instead
+                    myRasterLayerName = oneGeoPackageRaster.Path.Split('/').Last();
+                }
+
+                // Append the 'type of layer' to the myRasterLayerName string to display in the 
+                // ListBox and as the key for the HybridDictonary
+                myRasterLayerName = myRasterLayerName + " - RasterLayer";
+
+                // Add the name of the RasterLayer and the RasterLayer itself into the HybridDictionary
+                _MyHybridDictionary_Layers.Add(myRasterLayerName, myRasterLayer);
+
+                // Add the name of the RasterLayer to ObservableCollection _MyLayerNamesNotInTheMap 
+                // which displays the human-readable layer names used by the _myListView_LayersNotInTheMap
+                _MyLayerNamesNotInTheMap.Add(myRasterLayerName);
+            }
+
+            // Get the read only list of GeoPackageFeatureTabless from the GeoPackage
+            IReadOnlyList<GeoPackageFeatureTable> myReadOnlyListOfGeoPackageFeatureTables = myGeoPackage.GeoPackageFeatureTables;
+
+            // Loop through each GeoPackageFeatureTable
+            foreach (GeoPackageFeatureTable oneGeoPackageFeatureLayer in myReadOnlyListOfGeoPackageFeatureTables)
```

Comment:
Naming for `oneGeoPackageFeatureLayer` might be confusing - I think this makes it seem like there is already a layer, but it is actually a table. E.g. `new FeatureLayer(oneGeoPackageFeatureLayer)` implies that the FeaturLayer constructor takes a layer.